packages/myst-ext-button/src/index.ts

```diff
@@ -1,28 +1,72 @@
 import type { RoleSpec, RoleData, GenericNode } from 'myst-common';
 import type { Link } from 'myst-spec-ext';
+import { fileWarn, RuleId } from 'myst-common';
 
-const REF_PATTERN = /^(.+?)<([^<>]+)>$/;
+// Matches "text<link>" capturing body text (group 1) and an optional "<link>" suffix (group 2).
+// Group 2 keeps the angle brackets; it can also be exactly "<>" to signal an empty target.
+const TEXT_LINK_PATTERN = /^([^<>]*)(<[^<>]*>)?$/;
 
 export const buttonRole: RoleSpec = {
   name: 'button',
-  doc: 'Button element with an action to navigate to internal or external links.',
+  doc: 'Button element with an action to navigate to internal or external links. See [](#ui:buttons).',
```

Comment:
This is a little addition to make it easier to find the more verbose docs about this from the reference page. I opened an issue to do this for others here:

- https://github.com/jupyter-book/mystmd/issues/2609